web-registry/src/components/organisms/ChooseCreditClass/ClassOptions.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState, useEffect } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
+import { Grid } from '@mui/material';
+
+import { ChooseClassTemplate } from './ChooseClassTemplate';
+import { ClassOption } from './ClassOption';
+import { useGetCreditClassOptions } from './hooks/useGetCreditClassOptions';
+import { validate, getProjectPageBaseData } from '../../../lib/rdf';
+import {
+  useUpdateProjectByIdMutation,
+  useShaclGraphByUriQuery,
+} from '../../../generated/graphql';
+
+const ClassOptions: React.FC = () => {
+  const [error, setError] = useState('');
+  const navigate = useNavigate();
+  const { projectId } = useParams();
+  const { creditClassOptions, loading } = useGetCreditClassOptions();
+  const { data: graphData } = useShaclGraphByUriQuery({
+    variables: {
+      uri: 'http://regen.network/ProjectPageShape',
+    },
+  });
+  const [updateProject] = useUpdateProjectByIdMutation();
+
+  // Reset error message, so it reappears when user tries again
+  useEffect(() => {
+    if (error) {
+      setTimeout(() => {
+        setError('');
+      }, 5000);
+    }
+  }, [error]);
+
+  async function handleSelection(
+    creditClassId: string,
+    creditClassOnChainId: string,
+  ): Promise<void> {
+    if (graphData?.shaclGraphByUri?.graph) {
+      const metadata = getProjectPageBaseData(creditClassOnChainId);
+
+      try {
+        const report = await validate(
+          graphData.shaclGraphByUri.graph,
+          metadata,
+          'http://regen.network/ProjectPageCreditClassGroup',
+        );
+        if (report.conforms) {
+          try {
+            await updateProject({
+              variables: {
+                input: {
+                  id: projectId,
+                  projectPatch: {
+                    creditClassId,
```

Comment:
if there's no corresponding off-chain credit class, we shouldn't provide this to the projectPatch because this results in some invalid UUID error: `Variable \"$input\" got invalid value \"\" at \"input.projectPatch.creditClassId\"; Expected type \"UUID\". Invalid UUID...`

Reply:
fixed